client/state/sites/reducer.js

```diff
@@ -125,24 +127,60 @@ export function items( state = {}, action ) {
 			};
 		}
 
-		case SITE_ICON_SET: {
-			const { siteId, iconUrl } = action;
+		case SITE_SETTINGS_UPDATE:
+		case SITE_SETTINGS_RECEIVE: {
```

Comment:
>So, If I understand properly we're receiving the site Icon ID from the site settings endpoint, I'm wondering why we're storing it in this reducer, instead of relying on the siteSettings state to handle it (like other settings)?
>Nevermind, after review the API diff, I understand why you had to do that :)

Yeah, with these changes, we're treating `site.icon.media_id` as the preferred source of truth for the site icon, which means we want to reconcile when incoming settings are received. Now that you mention it though, this might need to go both ways: when we receive site(s), and we know those sites in the `siteSettings` reducer, we'll probably want to update the site icon value there (though this likely would have little effect since, as I mentioned, we want to prefer using it on the site state anyways).